crates/common/mqtt_channel/src/connection.rs

```diff
@@ -117,10 +119,9 @@ impl Connection {
                         return Err(err);
                     };
                     let subscriptions = config.subscriptions.filters();
-                    if subscriptions.is_empty() {
+                    if !Connection::subscribe_to_topics(&mqtt_client, subscriptions).await {
                         break;
                     }
-                    mqtt_client.subscribe_many(subscriptions).await?;
```

Comment:
The behavior must no change on error.

```suggestion
                    Connection::subscribe_to_topics(&mqtt_client, subscriptions).await?;
```

Reply:
I refactored to return the error now.